sw/device/lib/dif/dif_rv_timer.c

```diff
@@ -0,0 +1,191 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_rv_timer.h"
+
+#include <stddef.h>
+
+#include "rv_timer_regs.h"  // Generated.
+
+#define HART_CFG_ADDR_GAP 0x100
+
+dif_rv_timer_init_result_t dif_rv_timer_init(dif_rv_timer_config_t config,
+                                             dif_rv_timer_t *timer_out) {
+  if (timer_out == NULL) {
+    return kDifRvTimerInitPreconditionFailed;
+  }
+
+  uint32_t cfg_value = 0;
+  cfg_value |= (config.tick_params.prescale & RV_TIMER_CFG0_PRESCALE_MASK)
+               << RV_TIMER_CFG0_PRESCALE_OFFSET;
+  cfg_value |= (config.tick_params.tick_step & RV_TIMER_CFG0_STEP_MASK)
+               << RV_TIMER_CFG0_STEP_OFFSET;
+  mmio_region_write32(
+      config.base_addr,
+      RV_TIMER_CFG0_REG_OFFSET + config.hart * HART_CFG_ADDR_GAP, cfg_value);
+
+  timer_out->base_addr = config.base_addr;
+  timer_out->hart = config.hart;
+
+  return kDifRvTimerInitOk;
+}
+
+// A simple implementation of the Euclidean Algorithm by repeated subtraction.
+static uint64_t euclidean_gcd(uint64_t a, uint64_t b) {
+  while (a != b) {
+    if (a > b) {
+      a -= b;
+    } else {
+      b -= a;
+    }
+  }
+
+  return a;
+}
+
+#define UINT12_MAX 0xfff
```

Comment:
I recommend using `RV_TIMER_CFG0_PRESCALE_MASK` instead of duplicating that value here. Things can easily break if this mask changes.

Reply:
Good observation.